src/CordovaLogger.js

```diff
@@ -17,204 +17,229 @@
+        if (isVerbose) {
+            message = error.stack;
+        } else {
+            message = error.message;
+        }
+    } else {
+        // Plain text error message
+        message = error;
+    }
+
+    if (typeof message === 'string' && !message.toUpperCase().startsWith('ERROR:')) {
+        // Needed for backward compatibility with external tools
+        message = 'Error: ' + message;
+    }
+
+    return message;
 }
 
 /**
- * Static method to create new or acquire existing instance.
- *
- * @return  {CordovaLogger}  Logger instance
+ * @typedef {'verbose'|'normal'|'warn'|'info'|'error'|'results'|string} CordovaLoggerLevel
  */
-CordovaLogger.get = function () {
-    return INSTANCE || (INSTANCE = new CordovaLogger());
-};
-
-CordovaLogger.VERBOSE = 'verbose';
-CordovaLogger.NORMAL = 'normal';
-CordovaLogger.WARN = 'warn';
-CordovaLogger.INFO = 'info';
-CordovaLogger.ERROR = 'error';
-CordovaLogger.RESULTS = 'results';
 
 /**
- * Emits log message to process' stdout/stderr depending on message's severity
- *   and current log level. If severity is less than current logger's level,
- *   then the message is ignored.
- *
- * @param   {String}  logLevel  The message's log level. The logger should have
- *   corresponding level added (via logger.addLevel), otherwise
- *   `CordovaLogger.NORMAL` level will be used.
- * @param   {String}  message   The message, that should be logged to process'
- *   stdio
+ * Implements logging facility that anybody could use.
  *
- * @return  {CordovaLogger}     Current instance, to allow calls chaining.
+ * Should not be instantiated directly! `CordovaLogger.get()` method should be
+ * used instead to acquire the logger instance.
  */
-CordovaLogger.prototype.log = function (logLevel, message) {
-    // if there is no such logLevel defined, or provided level has
-    // less severity than active level, then just ignore this call and return
-    if (!this.levels[logLevel] || this.levels[logLevel] < this.levels[this.logLevel]) {
-        // return instance to allow to chain calls
-        return this;
-    }
-
-    var isVerbose = this.logLevel === 'verbose';
-    var cursor = this.stdoutCursor;
-
-    if (message instanceof Error || logLevel === CordovaLogger.ERROR) {
-        message = formatError(message, isVerbose);
-        cursor = this.stderrCursor;
+class CordovaLogger {
+    // Encapsulate the default logging level values with constants:
+    static get VERBOSE () { return 'verbose'; }
+    static get NORMAL () { return 'normal'; }
+    static get WARN () { return 'warn'; }
+    static get INFO () { return 'info'; }
+    static get ERROR () { return 'error'; }
+    static get RESULTS () { return 'results'; }
+
+    /**
+     * Static method to create new or acquire existing instance.
+     *
+     * @returns {CordovaLogger} Logger instance
+     */
+    static get () {
+        if (Object.getOwnPropertySymbols(global).indexOf(INSTANCE_KEY) === -1) {
```

Comment:
It's the global object. In a web browser, it would be `window`:
https://nodejs.org/dist/latest-v10.x/docs/api/globals.html#globals_global

Reply:
Got it. I would really appreciate pointers for some newer features such as `global` object & `Symbol` in the source or _maybe_ in the commit message.

Reply:
`global` is a NodeJS specific thing, which I think has existed since the very early days.
`Symbol` is a built-in JS (ES6) language feature, which has been around for several years now.

I don't really think we should have comments everywhere providing an introduction to built-in language features, but I think it is probably valuable to have a link to https://derickbailey.com/2016/03/09/creating-a-true-singleton-in-node-js-with-es6-symbols/ somewhere to explain the singleton design pattern we're implementing. I'll amend the commit with that added.